src/EPiServer.SocialAlloy.Web/Social/Repositories/Moderation/CommunityMembershipModerationRepository.cs

```diff
@@ -127,18 +131,23 @@ private AddMemberRequest GetMembershipRequest(string userId, string communityId)
         /// <param name="user">The user under moderation</param>
         /// <param name="group">The group that membership is being moderated</param>
         /// <returns>composite of WorkflowItem and AddMemberRequest</returns>
-        private Composite<WorkflowItem, AddMemberRequest> GetComposite(string user, string group)
+        private WorkflowItem<AddMemberRequest> GetComposite(string user, string group)
         {
-            Composite<WorkflowItem, AddMemberRequest> memberRequest = null;
+            // Construct the filters to apply to get the desired target under moderation
+            var filters = new List<FilterExpression>();
 
-            //Construct a filter to return the desired target under moderation
-            var filter = new CompositeCriteria<WorkflowItemFilter, AddMemberRequest>();
-            filter.Filter.Target = Reference.Create(CreateUri(group, user));
+            filters.Add(this.workflowitemFilters.Target.EqualTo(Reference.Create(CreateUri(group, user))));
+            filters.Add(this.workflowitemFilters.Extension.Type.Is<AddMemberRequest>());
+
+            var filter = new Criteria
+            {
+                Filter = new AndExpression(filters)
+            };
 
             try
             {
                 //retrieve the first workflow that matches the target filter 
```

Comment:
Minor: Please add a space after // and capitalize first letter in "retrieve" to match our comment styling elsewhere. thanks.